RealmSwift/Impl/Persistable.swift

```diff
@@ -51,7 +51,7 @@ public protocol _Persistable: _RealmSchemaDiscoverable, _ObjcBridgeable {
     static func _rlmKeyPathRecorder(with lastAccessedNames: NSMutableArray) -> Self
     // The type which is actually stored in the Realm. This is Self for types
     // we support directly, but may be a different type for enums and mapped types.
-    associatedtype _RealmValue: _Persistable
+    associatedtype PersistedType: _Persistable
```

Comment:
This renaming is aligning it to what the public API for custom type mappings will use. It's not strictly part of this PR but renaming everything back to _RealmValue would have just produced annoying merge conflicts.